service/src/main/java/dk/dbc/opensearch/cache/HttpFetcher.java

```diff
@@ -98,15 +111,92 @@ public Context with(String key, String value) {
+                log.debug("Error fetching resource: ", ex);
+                throw new UserMessageException(UserMessage.BAD_RESPONSE);
+            }
+            if (is == null) {
+                log.error("Error fetching resource: {}: No content");
+                throw new UserMessageException(UserMessage.BAD_RESPONSE);
+            }
+            return is;
+        }
+    }
+
+    public final static class PostContext {
+
+        private final Client client;
+        private final UriBuilder uriBuilder;
+        private final HashMap<String, String> values;
+        private final String userAgent;
+
+        private PostContext(Client client, UriBuilder uriBuilder, String userAgent, String trackingId) {
+            this.client = client;
+            this.uriBuilder = uriBuilder;
+            this.values = new HashMap<>();
+            this.userAgent = userAgent;
+            this.values.put("trackingId", trackingId);
+        }
+
+        /**
+         * Set path parameters (substitution)
+         *
+         * @param key   name of the parameter
+         * @param value what to place instead
+         * @return self for chaining
+         */
+        public PostContext with(String key, String value) {
+            values.put(key, value);
+            return this;
+        }
+
+        /**
+         * Get the request as an InputStream
+         *
+         * @param entity the entity to post
+         * @param stats  Statistics module
+         * @param name   request name for statistics
+         * @return InputStream and throws runtime exception in case of an error
+         */
+        public InputStream request(Entity entity, StatisticsRecorder stats, String name) {
+            return request(entity, MediaType.APPLICATION_JSON_TYPE, stats, name);
+        }
+
+        /**
+         * Get the request as an InputStream
+         *
+         * @param entity     the entity to post
+         * @param acceptType the content type wanted from the remote server
+         * @param stats      Statistics module
+         * @param name       request name for statistics
+         * @return InputStream and throws runtime exception in case of an error
+         */
+        public InputStream request(Entity entity, MediaType acceptType, StatisticsRecorder stats, String name) {
             InputStream is = null;
             URI uri = uriBuilder.buildFromMap(values);
-            log.debug("Fetching: {}", uri);
-            try (Timing timer = stats.timer(name)) {
+            log.debug("Fetching: (POST) {}", uri);
+            try (final Timing timer = stats.timer(name)) {
                 is = client.target(uri)
-                        .request()
-                        .accept(MediaType.APPLICATION_JSON_TYPE)
-                        .header("User-Agent", userAgent)
-                        .get(InputStream.class);
+                        .request().accept(acceptType).header("User-Agent", userAgent).post(entity, InputStream.class);
```

Comment:
```suggestion
                        .request()
                        .accept(acceptType)
                        .header("User-Agent", userAgent)
                        .post(entity, InputStream.class);
```
Mangler vist lige at trykke enter?

Reply:
Forbandede IDE's refactor.